src/DebugEngineHost.Common/HostLogChannel.cs

```diff
@@ -0,0 +1,134 @@
+        /// <summary>
+        /// Logs that contain the most detailed messages.
+        /// These messages may contain sensitive application data.
+        /// These messages are disabled by default and should never be enabled in a production environment.
+        /// </summary>
+        Trace,
+        /// <summary>
+        /// Logs that are used for interactive investigation during development.
+        /// These logs should primarily contain information useful for debugging and have no long-term value.
+        /// </summary>
+        Debug,
+        /// <summary>
+        /// Logs that track the general flow of the application.
+        /// These logs should have long-term value.
+        /// </summary>
+        Information,
+        /// <summary>
+        /// Logs that highlight an abnormal or unexpected event in the application flow, but do not otherwise cause the application execution to stop.
+        /// </summary>
+        Warning,
+        /// <summary>
+        /// Logs that highlight when the current flow of execution is stopped due to a failure.
+        /// These should indicate a failure in the current activity, not an application-wide failure.
+        /// </summary>
+        Error,
+        /// <summary>
+        /// Logs that describe an unrecoverable application or system crash, or a catastrophic failure that requires immediate attention.
+        /// </summary>
+        Critical,
+        /// <summary>
+        /// Not used for writing log messages.
+        /// Specifies that a logging category should not write any messages.
+        /// </summary>
+        None
+    }
+
+    // This must match the interface in DebugEngineHost.ref.cs
+    public interface ILogChannel
+    {
+        void WriteLine(LogLevel level, string message);
+
+        void WriteLine(LogLevel level, string format, params object[] values);
+
+        void Flush();
+
+        void Close();
+    }
+
+    public class HostLogChannel : ILogChannel
+    {
+        private readonly Action<string> _log;
+        private readonly StreamWriter _logFile;
+        private LogLevel _logLevel;
+
+        // For Testsing only
+        internal HostLogChannel() { }
+
+        public HostLogChannel(Action<string> logAction, string file, LogLevel logLevel)
+        {
+            _log = logAction;
+
+            if (!string.IsNullOrEmpty(file))
+            {
+                _logFile = File.CreateText(file);
+            }
+
+            _logLevel = logLevel;
+        }
+
+        public void SetLogLevel(LogLevel level)
+        {
+            _logLevel = level;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="verbosity"></param>
+        /// <param name="message"></param>
+        public void WriteLine(LogLevel level, string message)
+        {
+            if (level >= _logLevel)
+            {
+                string levelMsg = string.Format(CultureInfo.InvariantCulture, "[{0}] {1}", level.ToString(), message);
```

Comment:
>string [](http://example.com/codeflow?start=16&length=6)

If we wind up using simplified logging levels, I would suggest only adding the prefix if the level isn't verbose.